sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -42,8 +42,7 @@ public static SetAnalyzedStatement analyze(SetStatement node) {
 
         if (!SetStatement.Scope.GLOBAL.equals(node.scope())) {
             Assignment assignment = node.assignments().get(0);
-            // parser does not allow using the parameter expressions as setting names in the SET statements,
-            // therefore it is fine to convert the expression to string here.
+            assignment = rewriteOldSettings(assignment);
```

Comment:
this setting to rewrite would never exists when reaching this branch, its only for non GLOBAL set statements like set session